src/index.ts

```diff
@@ -191,35 +192,53 @@ export class Onfirework<T> {
    * @memberof Onfirework
    * @see https://firebase.google.com/docs/firestore/query-data/queries
    */
-  listDocs(filter?: Filter<T>[], limit?:number): Promise<Result<T>[]> {
-    return new Promise((resolve, reject) => {
-      let call:DocumentData = this.db.collection(this.collection);
-      if (filter) filter.map((data: Filter<T>) => {
+  async listDocs(filter?: Filter<T>[], limit?:number): Promise<Result<T>[]> {
+    let call:DocumentData = this.db.collection(this.collection);
+    const rangeOperators = ['<', '<=', '>', '>='];
+    const rangeFilters: Filter<T>[] = [];
+    if (filter) filter.forEach((data: Filter<T>) => {
+      if(rangeOperators.includes(data[1])) {
+        rangeFilters.push(data);
+      } else {
         call = call.where(...data);
-        return call;
-      });
-      if (limit) call = call.limit(limit)
+      }
+    });
+    if (limit) call = call.limit(limit)
```

Comment:
With the new approach, if the filters contain some range operators the limit may not work properly

Reply:
The limit must be AFTER child queries